api/lib/domain/events/handle-campaign-participation-results-sending.js

```diff
@@ -8,77 +8,106 @@ const PAYLOAD_CAMPAIGN_TYPE = 'EVALUATION';
 const PAYLOAD_STRUCTURE_NAME = 'Pix';
 const PAYLOAD_STRUCTURE_TYPE = 'externe';
 const PAYLOAD_CAMPAIGN_URL = `${config.domain.pixApp}${config.domain.tldFr}/campagnes`;
+const PAYLOAD_TEST_STATE = { STARTED: 2, FINISHED: 3, SENT: 4 };
+const PAYLOAD_UNITS = { PERCENTAGE: 'A', SCORE: 'B' };
+const PAYLOAD_EVALUATION_CATEGORY = 'competence';
+const PAYLOAD_EVALUATION_TYPE = 'competence Pix';
+const PAYLOAD_PROGRESSION = { FINISHED: 100 };
+
+function getTypeTest(targetProfileName) {
+  if (targetProfileName.includes('Diagnostic initial')) {
+    return 'DI';
+  } else if (targetProfileName.includes('Parcours complet')) {
+    return 'PC';
+  }
+  return 'CP';
+}
+
+class PoleEmploiPayload {
+  constructor({ user, campaign, targetProfile, participation, participationResult }) {
+    this.individu = {
+      nom: user.lastName,
+      prenom: user.firstName,
+    };
+
+    this.campagne = {
+      nom: campaign.name,
+      dateDebut: campaign.createdAt,
+      dateFin: campaign.archivedAt,
+      type: PAYLOAD_CAMPAIGN_TYPE,
+      idCampagne: campaign.id,
+      codeCampagne: campaign.code,
+      urlCampagne: `${PAYLOAD_CAMPAIGN_URL}/${campaign.code}`,
+      nomOrganisme: PAYLOAD_STRUCTURE_NAME,
+      typeOrganisme: PAYLOAD_STRUCTURE_TYPE,
+    };
+
+    this.test = {
+      etat: PAYLOAD_TEST_STATE.SENT,
+      progression: PAYLOAD_PROGRESSION.FINISHED,
+      typeTest: getTypeTest(targetProfile.name),
+      referenceExterne: participation.id,
+      dateDebut: participation.createdAt,
+      dateProgression: participation.sharedAt,
+      dateValidation: participation.sharedAt,
+      evaluationCible: participationResult.masteryPercentage,
+      uniteEvaluation: PAYLOAD_UNITS.PERCENTAGE,
+      elementsEvalues: participationResult.competenceResults.map((competence) => ({
+        libelle: competence.name,
+        categorie: PAYLOAD_EVALUATION_CATEGORY,
+        type: PAYLOAD_EVALUATION_TYPE,
+        domaineRattachement: competence.areaName,
+        nbSousElements: competence.totalSkillsCount,
+        evaluation: {
+          scoreObtenu: competence.masteryPercentage,
+          uniteScore: PAYLOAD_UNITS.PERCENTAGE,
+          nbSousElementValide: competence.validatedSkillsCount,
+        },
+      })),
+    };
+  }
```

Comment:
**suggestion (non-blocking) :** Extraire ce bout de code dans une fonction privée
+ extraire une deuxième fonction privée pour la partie `elementEvalues` qui serait appelée dans la fonction privée  pour construire `this.test`

```js
 elementsEvalues: participationResult.competenceResults.map((competence) => ({
        libelle: competence.name,
        categorie: PAYLOAD_EVALUATION_CATEGORY,
        type: PAYLOAD_EVALUATION_TYPE,
        domaineRattachement: competence.areaName,
        nbSousElements: competence.totalSkillsCount,
        evaluation: {
          scoreObtenu: competence.masteryPercentage,
          uniteScore: PAYLOAD_UNITS.PERCENTAGE,
          nbSousElementValide: competence.validatedSkillsCount,
        },
      })),
    };
```

Reply:
Done